src/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -2169,6 +2169,12 @@ private bool PrepareToPushMessageToBridge<T>(Message message, ResultProcessor<T>
             {
                 // Infer a server automatically
                 server = SelectServer(message);
+
+                // If we didn't find one successfully, and we're allowed, queue for any viable server
```

Comment:
If I am understand it correctly, I think it will be interesting to test this logic with a clustered cache. In case of a failure, the logic will select any viable server. This viable server might return a moved exception and the message will get redirected to the right endpoint. It should be fine if eventually it gets to the right endpoint. Does that sound correct?

Reply:
Correct! Let's make sure in tests though - I'll add some cluster tests around this to the PR.

Reply:
Added additional testing in 2c2b0ea51c406fecd765f851a96acbbce5f06f2d!